indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -104,6 +105,10 @@ public final static InputRow parseInputRow(Object value, InputRowParser parser)
     if(parser instanceof StringInputRowParser && value instanceof Text) {
       //Note: This is to ensure backward compatibility with 0.7.0 and before
       return ((StringInputRowParser) parser).parse(value.toString());
+    } else if(parser instanceof StringInputRowParser && value instanceof BytesWritable) {
```

Comment:
maybe we could have a level of indirection where you specify ```parserClass``` rather than an actual ```parser```. I'm not sure if that's better or worse than the other alternatives on the table.